tests/device_tests/test_msg_authorize_coinjoin.py

```diff
@@ -0,0 +1,417 @@
+        ),
+    )
+    inp2 = messages.TxInputType(
+        address_n=parse_path("84'/1'/0'/1/0"),
+        amount=7289000,
+        prev_hash=TXHASH_65b811,
+        prev_index=1,
+        script_type=messages.InputScriptType.SPENDWITNESS,
+    )
+
+    # Other's coinjoined output.
+    out1 = messages.TxOutputType(
+        address="tb1qk7j3ahs2v6hrv4v282cf0tvxh0vqq7rpt3zcml",
+        amount=30000,
+        script_type=messages.OutputScriptType.PAYTOWITNESS,
+    )
+    # Other's coinjoined output.
+    out2 = messages.TxOutputType(
+        address="tb1q9cqhdr9ydetjzrct6tyeuccws9505hl96azwxk",
+        amount=30000,
+        script_type=messages.OutputScriptType.PAYTOWITNESS,
+    )
+    # Our coinjoined output.
+    out3 = messages.TxOutputType(
+        address_n=parse_path("84'/1'/0'/1/1"),
+        amount=50000,
+        script_type=messages.OutputScriptType.PAYTOWITNESS,
+    )
+    # Our coinjoined output.
+    out4 = messages.TxOutputType(
+        address_n=parse_path("84'/1'/0'/1/2"),
+        amount=50000,
+        script_type=messages.OutputScriptType.PAYTOWITNESS,
+    )
+    # Our change output.
+    out5 = messages.TxOutputType(
+        address_n=parse_path("84'/1'/0'/1/2"),
+        amount=7289000 - 50000 - 50000 - 10 - 5000,
+        script_type=messages.OutputScriptType.PAYTOWITNESS,
+    )
+    # Other's change output.
+    out6 = messages.TxOutputType(
+        address="tb1q9cqhdr9ydetjzrct6tyeuccws9505hl96azwxk",
+        amount=100000 - 30000 - 30000 - 6 - 5000,
+        script_type=messages.OutputScriptType.PAYTOWITNESS,
+    )
+    # Coordinator's output.
+    out7 = messages.TxOutputType(
+        address="mvbu1Gdy8SUjTenqerxUaZyYjmveZvt33q",
+        amount=16,
+        script_type=messages.OutputScriptType.PAYTOWITNESS,
+    )
+
+    with pytest.raises(TrezorFailure, match="No anonymity gain"):
+        btc.sign_tx(
+            client,
+            "Testnet",
+            [inp1, inp2],
+            [out1, out2, out3, out4, out5, out6, out7],
+            prev_txes=TX_CACHE_TESTNET,
+            preauthorized=True,
+        )
+
+
+def test_wrong_coordinator(client):
+    # Ensure that a preauthorized GetOwnershipProof fails if the commitment_data doesn't match the coordinator.
+
+    btc.authorize_coinjoin(
+        client,
+        max_total_fee=50000,
+        coordinator="www.example.com",
+        n=parse_path("m/84'/1'/0'"),
+        coin_name="Testnet",
+        script_type=messages.InputScriptType.SPENDWITNESS,
+    )
+
+    with pytest.raises(TrezorFailure, match="Unauthorized operation"):
+        ownership_proof, _ = btc.get_ownership_proof(
+            client,
+            "Testnet",
+            parse_path("84'/1'/0'/1/0"),
+            script_type=messages.InputScriptType.SPENDWITNESS,
+            user_confirmation=True,
+            commitment_data=b"www.example.org" + (1).to_bytes(ROUND_ID_LEN, "big"),
+            preauthorized=True,
+        )
+
+
+def test_cancel_authorization(client):
```

Comment:
we should have a test for multi-session authorization

i.e. in session 1 authorize for example.org
in session 2 authorize for example.com
switch back to 1, check that the right authorization is up
etc.
and similar for `cancel_authorization`

i'm ok with deferring this to a new separate issue if you don't want to delay this

Reply:
All done, https://github.com/trezor/trezor-firmware/pull/1127/commits/a0069eaebd4dedd5274ccf155fb3d2413b56c2c7..